### DISCUSSION TITLE
Tautulli automatic authentication, neither basic HTTP nor Plex working

### DISCUSSION THREAD

**Discussion by mgrimace** [goauthentik/authentik#6069] [Q&A]
Hello,

I'm trying to Tautulli working with either Plex authentication or basic http authentication per the official Authentik instructions [here](https://goauthentik.io/integrations/services/tautulli/). I also use NPM as my reverse proxy with Cloudflare SSL. My goal is to only have the Authentik sign-on, which automatically logs into Tautulli. Neither Plex nor basic HTTP per the guide are working. 

### Basic HTTP authentication

![](https://d33wubrfki0l68.cloudfront.net/2e9e1c63bf511433214020953aeb13fb44414eed/e7911/assets/images/tautulli-5a22d59b71995f7078716de87f258fd8.png)

This feature has moved to the config for Tautulli and is no longer available in the UI. However, it can be enabled manually. With basic HTTP authentication enabled in the config, after logging in to Authentik I'm passed to a Tautulli screen that still prompts for user/password. 

### Passing through a Plex login token
I got this working with Overseer using the following custom scope mapping. I tried duplicating this for Tautulli (changing the base URL) and no go.

```
from authentik.sources.plex.models import PlexSourceConnection
import json

connection = PlexSourceConnection.objects.filter(user=request.user).first()
if not connection:
    ak_logger.info("PLEX: No Plex connection found")
    return {}

base_url = "http://overseerr:5055"
end_point = "/api/v1/auth/plex"
token = "${overseer_token}"

headers = {
    "Content-Type": "application/json",
}

data = {
    "authToken": connection.plex_token
}


response = requests.post(base_url + end_point,
                         headers=headers, data=json.dumps(data))

if (response.status_code == 200):

    sid_value = response.cookies.get("connect.sid")
    cookie_obj = f"connect.sid={sid_value}"
    ak_logger.info("PLEX: The request was a success!")
    return {
        "ak_proxy": {
            "user_attributes": {
                "additionalHeaders": {
                    "Set-Cookie": cookie_obj
                }
            }
        }
    }
else:
    ak_logger.info(f"PLEX: The request failed with: {response.text}")
    return {}
```

As it is, does anyone have suggestions to avoid the duplicate login for Tautulli with Authentik?

**Comment by TDX44** [comment]
Any update to this? I've been stuck here for a week.

**Comment by jacoknapp** [comment]
So it looks like by default the password is now being hashed. This means that basic auth wont work. In order to make it work shutdown tautulli then go into the config.ini set the http_hash_password = 0, set http_password = Your password and then set http_basic_auth = 1, then start up tautulli again. This worked for me. Hopefully the maintainer sees this, and can put the feature options back into the panel.

**Reply by TDX44** [reply]
Thanks, I'll look into this tonight and update.

**Reply by camwoodhouse** [reply]
Hey, did you ever to this to work?
I've followed the instructions by @jacoknapp and edited by config.ini file, updated my Authentik Provider with basic HTTP Authentication and inputted the username and password in the Authentik Group, but no luck as it still presents me with the Tautulli basic HTTP login screen once I've logged into Authentik.

**Reply by dsteoh** [reply]
Had the same issue until I've rtfm. If you are using forward auth. Make sure to double check you are forwarding authorization header. Authentik's Caddy example, you'll have to add `authorization` to copy_headers
```
# capitalization of the headers is important, otherwise they will be empty
copy_headers authorization X-Authentik-User....
# Add the 'authorization' header to the list if you need proxy providers which
# send a custom HTTP-Basic Authentication header based on values from authentik
```
